packages/artifact_proxy/lib/artifact_proxy.dart

```diff
@@ -0,0 +1,87 @@
+// ignore_for_file: avoid_print
+
+import 'package:collection/collection.dart';
+import 'package:shelf/shelf.dart';
+
+/// A [Handler] that proxies artifact requests to the correct location.
+/// This is determined based on the [config].
+Handler artifactProxyHandler({required Map<dynamic, dynamic> config}) {
+  final engineMappings = config['engine_mappings'] as Map;
+  final shorebirdEngineRevisions = engineMappings.keys.cast<String>();
+
+  return (Request request) {
+    final path = request.url.path;
+    final shorebirdEngineRevision = shorebirdEngineRevisions.firstWhereOrNull(
+      path.contains,
+    );
+
+    final normalizedPath = shorebirdEngineRevision != null
+        ? path.replaceAll(shorebirdEngineRevision, r'$engine')
+        : path;
+
+    if (shorebirdEngineRevision == null) {
+      final location = getFlutterArtifactLocation(artifactPath: normalizedPath);
+      print('No engine revision detected, forwarding to: $location');
+      return Response.found(location);
+    }
+
+    final engineMapping = engineMappings[shorebirdEngineRevision] as Map;
+    final shorebirdOverrides =
+        engineMapping['shorebird_artifact_overrides'] as List;
+    final flutterEngineRevision =
+        engineMapping['flutter_engine_revision'] as String;
+    final shorebirdStorageBucket =
+        engineMapping['shorebird_storage_bucket'] as String;
```

Comment:
I suspect this could just default to something in the code and you can save yourself some typing in the config. 🤷

Reply:
I did this in case we move to a different storage bucket in the future but happy to remove it from the config